src/ol/render/canvas/Executor.js

```diff
@@ -270,11 +288,11 @@ class Executor {
    * @param {Array<*>} fillInstruction Fill instruction.
    * @param {Array<*>} strokeInstruction Stroke instruction.
    */
-  replayImage_(
+  replayImageOrLabel_(
```

Comment:
Oh.  Perhaps because of this.

Reply:
Yes indeed. A clear separation of the two code paths might make sense, but that would be a much more verbose change.